packages/flutter/lib/src/widgets/image.dart

```diff
@@ -240,6 +234,14 @@ class _ImageState extends State<Image> {
 
   @override
   Widget build(BuildContext context) {
+    // This one-time initialization could have been done in initState() since
+    // changes to the inherited widgets that _resolveImage depends on, notably
+    // DefaultAssetBundle, are handle by the dependenciesChanged() method. We're
+    // doing it here instead to avoid the assert that disallows references to
+    // inherited widgets at initState() time.
```

Comment:
"...because we have found that asserts catches bugs frequently, and is therefore worth this minor inconvenience here."